src/python/pants/engine/internals/parser.py

```diff
@@ -33,11 +33,13 @@ class HydratedStruct:
 
 class Parser(ABC):
     @abstractmethod
-    def parse(self, filepath: str, filecontent: bytes):
+    def parse(self, filepath: str, filecontent: bytes, extra_symbols: Dict[str, Any]):
```

Comment:
Ditto on `FrozenDict`. Or, if we don't care about the type safety, the more generic `Mapping` so that we can work with either `Dict` or `FrozenDict`.